### DISCUSSION TITLE
MemoryStream avoid autoplay when turn up

### DISCUSSION THREAD

**Discussion by joaquinmir** [pschatzmann/arduino-audio-tools#1275] [Q&A] [answered]
Hi! I'm using the memorystream class for playing a raw file when i press a button. So i wan't the files to not start automatically when i turn up the ESP32, i was trying with the end() function but didn't work. Here is the sketch: 
```
#include "AudioTools.h"
#include "hihat.h"
#include "kick.h"

AudioInfo info(44100, 1, 16);
I2SStream i2s;  // Output to I2S
MemoryStream music(hihat_raw, hihat_raw_len);
MemoryStream music2(kick_raw, kick_raw_len);
InputMixer<int16_t> mixer;
StreamCopy copier(i2s, mixer); // copies sound into i2s

void playButton(){
   music.begin();
   music2.begin();
}

void setup(){
    Serial.begin(115200);
    pinMode(18, INPUT_PULLDOWN);
    attachInterrupt(18, playButton, RISING);
    AudioLogger::instance().begin(Serial, AudioLogger::Info);
    mixer.add(music);
    mixer.add(music2);
    auto config = i2s.defaultConfig(TX_MODE);
    config.copyFrom(info);
    config.channels=1;
    config.sample_rate= 44100; 
    i2s.begin(config);
    music.end();
    music2.end();
}



void loop(){
  copier.copy();
}
```

**Comment by pschatzmann** [comment, marked as answer]
With the current implementation, adding copier.setActive(false) to the setup() and copier.setActive(true) to the playButton() should do the trick.

I  committed a change that sets [MemoryStream](https://pschatzmann.github.io/arduino-audio-tools/classaudio__tools_1_1_memory_stream.html) to be inactive when you call end(), so if you upgrade, your solution should work w/o any change as well or instead of calling end() you could set it to inactive in the constructor e.g. with MemoryStream music(hihat_raw, hihat_raw_len, __false__);

**Reply by joaquinmir** [reply]
Thank you!
